subxt/src/blocks/extrinsic_types.rs

```diff
@@ -215,6 +215,9 @@ where
 
         let metadata = client.metadata();
 
+        // removing the compact encoded prefix:
+        let extrinsic_bytes: Vec<u8> = Decode::decode(&mut &extrinsic_bytes[..])?;
```

Comment:
Here and in the `.position(..)` thing below, we could avoid the allocating by instead decoding the `Compact<u64>` off the front of the bytes and then seeing how many bytes were consumed in doing this.
(I say "avoid allocating" but this was copying the bytes into an `Arc` before, so either way you may allocate, but at least have the chocie of exactly where)

This "compact length" value (or in other words an "extrinsic_start_index" could be saved if useful in `ExtrinsicDetails` so that it's easy to get the rest of the bytes from it when needed.

I'd have to look at the code in more detail to see what the `Arc` was used for before and whether it's then good to keep it, or whether perhaps the `Vec<ChainBlockExtrinsic>` in `Extrinsics` could be instead something like `Arc<[ChainBlockExtrinsic]>` so that we don't allocate at all when we `iter()` over the extrinsics :)

Reply:
We could also give the `ExtrinsicDetails` an explicit lifetime that is bound by the lifetime of `Extrinsics`. This way we can degrade the bytes field of `ExtrinsicDetails` to `bytes: &[u8]` instead of `bytes: Arc<[u8]>` where the slice points to the `i..` bytes of the Vec<u8> of the respective ChainBlockExtrinsic, where i is the compact encoded length. This also avoids allocation. I am not sure though if the Arc was necessary somewhere.